lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1986,6 +2024,25 @@ llvm::Value *CodeGenModule::emitReturnDataCopyBytes(llvm::Value *DataOffset,
   return Bytes;
 }
 
+void CodeGenModule::emitReturnDataCopy(llvm::Value *ResultOffset,
+                                       llvm::Value *DataOffset,
+                                       llvm::Value *DataLength) {
+  if (isEVM()) {
+    auto DestOffset = Builder.CreatePtrToInt(ResultOffset, EVMIntTy);
+    auto Offset = Builder.CreateZExtOrTrunc(DataOffset, EVMIntTy);
+    auto Length = Builder.CreateZExtOrTrunc(DataLength, EVMIntTy);
+    Builder.CreateCall(Func_returnDataCopy, {DestOffset, Offset, Length});
+  } else if (isEWASM()) {
+    auto DestOffset = Builder.CreatePtrToInt(ResultOffset, Int8PtrTy);
```

Comment:
Should the `DestOffset` be `Int32PtrTy`?
https://ewasm.readthedocs.io/en/mkdocs/eth_interface/#returndatacopy

Reply:
There are some inconsistencies in the old implementation between the EEI definition and the API. For example, `call() ` defined [here](https://github.com/eee4017/SOLL/blob/frank-develop/lib/CodeGen/CodeGenModule.cpp#L448) use `Int8PtrTy`, `AddressPtrTy` instead of `Int32PtrTy` . Should all of them need to be modified so that they would be compatible with EEI?